### DISCUSSION TITLE
`gatsby-plugin-image` type errors

### DISCUSSION THREAD

**Discussion by noahsb** [gatsbyjs/gatsby#30660] [Help] [answered]
Hi! We're migrating to `gatsby-plugin-image` from `gatsby-image`. After following the instructions, I ran the codemod. About 32 files were modified successfully, but some resulted in the following errors:

1. `gatsby-plugin-image` was imported twice
2. I get type errors with this newly modified code (errors written as comments, some code redacted):

```
import { GatsbyImage } from "gatsby-plugin-image";

interface Props {
  data: {
    markdownRemark: {
      excerpt: string;
      frontmatter: {
        author?: string;
        date: Date;
        featuredImage?: {
          childImageSharp: {
            gatsbyImageData: any; // what is the correct type for this?
          }
          publicURL: string;
        };
        title: string;
        path: string;
      }
...
...
return (
...
            {featuredImageObject ? (
              <div className={styles.featuredImage}>
                <GatsbyImage image={featuredImageObject.childImageSharp.gatsbyImageData} alt={title} /> // is this the correct usage?
              </div>
            ) : (
...
);
...
export const pageQuery = graphql`query ($path: String!) {
  markdownRemark(frontmatter: {path: {eq: $path}}) {
    excerpt
    htmlAst
    frontmatter {
      author
      date(formatString: "MMMM DD, YYYY")
      featuredImage {
        childImageSharp {
          gatsbyImageData(layout: FULL_WIDTH) //is this acceptable?
        }
```

In summation, I want to find the correct type usage for `gatsbyImageData` above. Previously, I was using `fluid: ObjectFluid` but can't seem to find what to use without setting the type to `any` which I'd like to avoid.

Any suggestions on how to tackle this?

**Comment by oleksii-lukin** [comment, marked as answer]
I believe it should be `IGatsbyImageData`

```typescript
import { IGatsbyImageData } from 'gatsby-plugin-image'
```

and for layout look here https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-plugin-image/#layout

![image](https://user-images.githubusercontent.com/1281970/113587103-00757f00-9637-11eb-820e-a99657baa407.png)
```typescript
export declare type Layout = "fixed" | "fullWidth" | "constrained";
```

**Reply by noahsb** [reply]
This is exactly what I was looking for. Thank you!!

**Comment by ascorbic** [comment]
Hi. I'm a little unclear what the error is. Do you mean that it's complaining about `gatsbyImageData` being typed as `any`? @llamerr is correct, that the type is `IGatsbyImageData`.

**Reply by noahsb** [reply]
Hi. Yes, that's correct - thanks! I updated the discussion description so that this might be less confusing and more helpful for others with the same problem.
